tailscale/resource_dns_split_nameservers.go

```diff
@@ -24,6 +24,7 @@ func resourceDNSSplitNameservers() *schema.Resource {
 				Type:        schema.TypeString,
 				Description: "Domain to configure split DNS for. Requests for this domain will be resolved using the provided nameservers.",
 				Required:    true,
+				ForceNew:    true,
```

Comment:
Drive-by: I noticed that this provider uses `UpdateSplitDNS()`, which is a patch operation. This means that the resource is not setting all split DNS settings for the tailnet, but just those for a specific domain. As originally implemented, if the domain changes, the settings for the previous domain will be left alone. That seems confusing.

Using ForceNew here ensures that the settings for the previous domain are removed, so that the state of the system accurately reflects what's configured in Terraform.

Reply:
I'm going to wait for @mpminardi to chime in before merging, since this is a behavior change. From one perspective, it might be fixing buggy behavior, but possibly people could be relying on that buggy behavior ...

Reply:
Whoops, good catch! Agree that this is buggy / unintended behaviour here that we should fix. Feel like this should be a safe change since we are essentially leaving a dangling resource outside of Terraform's control / lifecycle otherwise.